client/blockchain-service/src/utils.rs

```diff
@@ -161,32 +161,52 @@ impl BlockchainService {
     ///
-    /// The provider IDs found are added to the [`BlockchainService`]'s list of provider IDs.
-    pub(crate) fn get_provider_ids(&mut self, block_hash: &H256) {
+    /// IMPORTANT! If there is more than one [`BCSV_KEY_TYPE`] key in this node's keystore, linked to
+    /// different Provider IDs, this function will panic. In other words, this node doesn't support
+    /// managing multiple Providers at once.
+    pub(crate) fn get_provider_id(&mut self, block_hash: &H256) {
+        let mut provider_ids_found = Vec::new();
         for key in self.keystore.sr25519_public_keys(BCSV_KEY_TYPE) {
-            self.client
+            let maybe_provider_id = match self
+                .client
                 .runtime_api()
                 .get_storage_provider_id(*block_hash, &key.into())
-                .map(|provider_id| {
-                    if let Some(provider_id) = provider_id {
-                        match provider_id {
-                            StorageProviderId::BackupStorageProvider(bsp_id) => {
-                                self.provider_ids.insert(bsp_id);
-                            }
-                            StorageProviderId::MainStorageProvider(msp_id) => {
-                                self.provider_ids.insert(msp_id);
-                            }
-                        }
-                    } else {
-                        warn!(target: LOG_TARGET, "There is no provider ID for key: {:?}. This means that the node has a BCSV key in the keystore for which there is no provider ID.", key);
-                    }
-                })
-                .unwrap_or_else(|_| {
-                    warn!(target: LOG_TARGET, "Failed to get provider ID for key: {:?}.", key);
-                });
+            {
+                Ok(provider_id) => provider_id,
+                Err(e) => {
+                    error!(target: LOG_TARGET, "Runtime API error while getting Provider ID for key: {:?}. Error: {:?}", key, e);
+                    continue;
+                }
+            };
+
+            match maybe_provider_id {
+                Some(provider_id) => {
+                    provider_ids_found.push(provider_id);
+                }
+                None => {
+                    debug!(target: LOG_TARGET, "There is no Provider ID for key: {:?}. This means that the node has a BCSV key in the keystore for which there is no Provider ID.", key);
+                }
+            };
+        }
+
+        // Case: There is no Provider ID linked to any of the [`BCSV_KEY_TYPE`] keys in this node's keystore.
+        // This is expected, if this node starts up before the Provider has been registered.
+        if provider_ids_found.is_empty() {
+            warn!(target: LOG_TARGET, "There is no Provider ID linked to any of the BCSV keys in this node's keystore. This is expected, if this node starts up before the BSP has been registered.");
```

Comment:
It is possible to start up a node without having yet signed up the BSP/MSP, so this should be possible. But in future PRs I'll try to make it so that if you start a node as BSP/MSP, but it has keys in its keystore for the opposite role, it will panic.